src/runtime_src/core/tools/xbmgmt2/flash/flasher.cpp

```diff
@@ -36,43 +36,54 @@
 #define FLASH_BASE_ADDRESS BPI_FLASH_OFFSET
 #define MAGIC_XLNX_STRING "xlnx" // from xclfeatures.h FeatureRomHeader
 
-Flasher::E_FlasherType Flasher::getFlashType(std::string typeStr)
-{
-    std::string err;
-    E_FlasherType type = E_FlasherType::UNKNOWN;
-    if (typeStr.empty())
-      typeStr = xrt_core::device_query<xrt_core::query::f_flash_type>(m_device);
-    if (typeStr.empty())
-      typeStr = xrt_core::device_query<xrt_core::query::flash_type>(m_device);
-
-    if (typeStr.empty())
-    {
-        getProgrammingTypeFromDeviceName(mFRHeader.VBNVName, type);
-    }
-    else if (typeStr.compare("spi") == 0)
-    {
+/* 
+ * map flash_tyoe str to E_FlasherType
+ */
+void Flasher::typeStr_to_E_FlasherType(const std::string& typeStr, E_FlasherType &type)
```

Comment:
This method should return back the enumeration type, as opposed to via a return variable.   Doing so allows the calling program to act upon the value instead of first creating a variable and then setting it.